src/diet.ts

```diff
@@ -1033,7 +1034,10 @@ export function runDiet(): void {
 
     MenuItem.defaultPriceFunction = (item: Item) => {
       const itemRetrievePrice = retrievePrice(item);
-      return itemRetrievePrice > 0 ? itemRetrievePrice : item.tradeable ? mallPrice(item) : 0;
+      const itemMallPrice = mallPrice(item) > 0 ? mallPrice(item) : Infinity;
+      const itemNpcPrice = npcPrice(item) > 0 ? npcPrice(item) : Infinity;
+      const bestPrice = Math.min(itemRetrievePrice, itemMallPrice, itemNpcPrice);
+      return bestPrice === Infinity && !item.tradeable && have(item) ? 0 : bestPrice; // Handle unpurchaseable items like distention pills
```

Comment:
I am (mildly) concerned that having unobtainable items with a price of `0` will result in bad behavior from the diet calculator; if you can point to a specific place where we prune that out I'll happily merge this now, but otherwise I'd like to wait for review from @pstalcup 

I know that right now there are situations where we have specific items with a price of 0, but I think a lot of those have bespoke limits in place (e.g., distention pill)